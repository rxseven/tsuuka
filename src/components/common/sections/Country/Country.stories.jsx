import React from 'react';
import { storiesOf } from '@storybook/react';

import Country from './index';

const title = 'Components/Common/Sections/Country';
const props = {
  data: [
    { code: 'ES', name: 'Spain' },
    { code: 'IT', name: 'Italy' },
    { code: 'TH', name: 'Thailand' }
  ]
};

storiesOf(title, module).add('default', () => <Country {...props} />);
