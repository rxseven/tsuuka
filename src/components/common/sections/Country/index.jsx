import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';
import styled from 'styled-components';

const API_URL = 'https://www.countryflags.io';

const Flag = styled.img`
  height: 16px;
  margin-right: 0.5rem;
  width: 16px;
`;

const Item = styled.li`
  align-items: center;
  display: flex;
  justify-content: flex-start;
  margin-bottom: 0.5rem;
`;

const List = styled.ul`
  list-style: none;
  margin: 0;
  padding-left: 0;
`;

const Name = styled.span`
  font-size: 0.875rem;
`;

const propTypes = exact({
  data: PropTypes.arrayOf(
    PropTypes.shape({
      code: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })
  ).isRequired
});

function Country({ data }) {
  return (
    <List>
      {data.map(country => (
        <Item key={country.code}>
          <Flag
            alt={country.name}
            src={`${API_URL}/${country.code}/flat/16.png`}
          />
          <Name>{country.name}</Name>
        </Item>
      ))}
    </List>
  );
}

Country.propTypes = propTypes;

export default Country;
