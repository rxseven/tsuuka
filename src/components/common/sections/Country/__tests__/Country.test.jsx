import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Country from '../index';

describe('<Country />', () => {
  // Arrange
  const props = {
    data: [
      { code: 'ES', name: 'Spain' },
      { code: 'IT', name: 'Italy' },
      { code: 'TH', name: 'Thailand' }
    ]
  };
  const component = <Country {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
