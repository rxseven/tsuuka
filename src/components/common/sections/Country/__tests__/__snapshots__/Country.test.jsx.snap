// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Country /> Snapshot tests should render correctly 1`] = `
<ul
  className="sc-htpNat eZEtUy"
>
  <li
    className="sc-bwzfXH hQcoUX"
  >
    <img
      alt="Spain"
      className="sc-bdVaJa gChJQj"
      src="https://www.countryflags.io/ES/flat/16.png"
    />
    <span
      className="sc-bxivhb hMCZnY"
    >
      Spain
    </span>
  </li>
  <li
    className="sc-bwzfXH hQcoUX"
  >
    <img
      alt="Italy"
      className="sc-bdVaJa gChJQj"
      src="https://www.countryflags.io/IT/flat/16.png"
    />
    <span
      className="sc-bxivhb hMCZnY"
    >
      Italy
    </span>
  </li>
  <li
    className="sc-bwzfXH hQcoUX"
  >
    <img
      alt="Thailand"
      className="sc-bdVaJa gChJQj"
      src="https://www.countryflags.io/TH/flat/16.png"
    />
    <span
      className="sc-bxivhb hMCZnY"
    >
      Thailand
    </span>
  </li>
</ul>
`;
